src/game_battlealgorithm.cpp

```diff
@@ -1004,7 +1004,7 @@ bool Game_BattleAlgorithm::Normal::Execute() {
 			}
 		}
 
-		if (effect > 0) {
+		if (Player::IsRPG2k() || effect > 0) {
```

Comment:
What is your basis for determining that 2k battles always add variance even if the damage is 0?
 
I'm looking at the 2k battle system in steam 2ke version in RE tools and in both normal attacks and skills I see it checking if the `damage > 0` before applying variance.

Are you using an older version of RPG_RT for your 2k testing? Which version are you using exactly?

Reply:
Sadly I cannot tell the exact version but it definitely is older than 1.50 (probably 1.10). I checked the RPG Maker 2000 changelog and for version 1.50 it is stated that special case with sometimes 1 damage despite calculated zero damage is fixed. So I have to change the check against RPG Maker 2000 versions before 1.50.

Reply:
if this is really only 1.5+ you can use https://github.com/EasyRPG/Player/blob/master/src/player.h#L180 ``IsRPG2kUpdated()``

Reply:
I have a bunch of RPG_RT binaries here, if you can compute a hash of the RPG_RT exe your'e using you can compare it to these to find out the version. You can use for example `md5sum` program on linux.

 https://github.com/fmatthew5876/rpgrt/tree/master/rpg2000

Reply:
Thank you for providing the binaries! I did some tests and can tell that this "sometimes 1 damage oddity" occurs in RPG Maker 2000 versions 1.05, 1.07, 1.09 and the RPG_RT provided by "Vampires Dawn". The versions 1.51 and 1.61 do not have this oddity. But I came across another problem: How does EasyRPG check the binary version? The console output always tells (even then when the RPG_RT binary is version 1.05, 1.07 or 1.09):
`Debug: RPG2k >= v1.50 / RPG2k3 >= v1.05 detected`
`Debug: Engine configured as: 2k=true 2k3=false MajorUpdated=true Eng=false`

Reply:
EasyRPG has a bunch of heuristics based on the LDB database and other things. We don't scan the RPG_RT.exe binary directly. Our heuristics are not perfect and sometimes get it wrong.

Thanks for this find. It indeed looks like it's possibly a change with older RPG_RT versions. I'll need to dig with RE tools to verify

Reply:
Added version check for RPG Maker 2000 version below 1.50 now.